src/shogun/classifier/GaussianProcessClassification.cpp

```diff
@@ -120,10 +120,8 @@ CBinaryLabels* CGaussianProcessClassification::apply_binary(
 		if (m_method->get_inference_type()== INF_FITC_LAPLACE_SINGLE)
 		{
 #ifdef USE_GPL_SHOGUN
-			CSingleFITCLaplaceInferenceMethod* fitc_method=
-				CSingleFITCLaplaceInferenceMethod::obtain_from_generic(m_method);
```

Comment:
`CSingleFITCLaplaceInferenceMethod::obtain_from_generic` method increases the reference counter, but your replacement doesnt.

Do you see what I mean here? You have written that you have dealt with this, but that is not true here

Reply:
ah **sorry**, you in fact remove the UNREF below ...
all good!